### PR TITLE
Vault integration: test helpers and token renewal

### PR DESCRIPTION
**Pull request by mgaffney**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -26,9 +26,11 @@ require (
 	github.com/hashicorp/go-kms-wrapping v0.6.1
 	github.com/hashicorp/go-multierror v1.1.1
 	github.com/hashicorp/go-retryablehttp v0.6.8
+	github.com/hashicorp/go-rootcerts v1.0.2 // indirect
```
Comment (louisruch) [inline go.mod:29]: `go mod tidy`

File: internal/credential/vault/vault.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+func newClient(c *clientConfig) (*client, error) {
+	const op = "vault.newClient"
+	if !c.IsValid() {
+		return nil, errors.New(errors.InvalidParameter, op, "invalid configuration")
+	}
+	vc := vault.DefaultConfig()
+	vc.Address = c.Addr
+	if len(c.CaCert) > 0 {
+		rootConfig := &rootcerts.Config{
+			CACertificate: []byte(c.CaCert),
```
Comment (louisruch) [inline internal/credential/vault/vault.go:51]: unnecessary type conversion `[]byte`

File: internal/credential/vault/vault.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		}
+		tlsConfig := vc.HttpClient.Transport.(*http.Transport).TLSClientConfig
```
Comment (louisruch) [inline internal/credential/vault/vault.go:53]: Since you have the `TlsSkipVerify` bool in the config would you not want to set`InsecureSkipVerify` here?

**Review by louisruch — APPROVED** [review]
Some comments but otherwise LGTM with a caveat: I did not do a extremely thorough review of the `vault_testing.go` changes it seems good and well tested, however I am not super familiar with setting up docker especially for vault, so I would not have caught any non-obvious mistakes...

File: go.mod (hunk 1 above)
Comment (mgaffney) [reply go.mod:29]: ty!

File: internal/credential/vault/vault.go (hunk 2 above)
Comment (mgaffney) [reply internal/credential/vault/vault.go:51]: ty!

File: internal/credential/vault/vault.go (hunk 3 above)
Comment (mgaffney) [reply internal/credential/vault/vault.go:53]: Great catch! Ty!

**Pull request merged**